api/src/lib.rs

```diff
@@ -2250,14 +2257,15 @@ pub extern "system" fn Java_com_horizen_commitmenttree_CommitmentTree_nativeAddS
         None
     };
 
-
-    let constant_fe;
+    // let constant_fe;
     let constant = if _constant_nullable.is_null() {
         Option::None
     } else {
         let constant_bytes = parse_jbyte_array_to_vec(&_env, &_constant_nullable, FIELD_SIZE);
-        constant_fe = FieldElement::deserialize(constant_bytes.as_slice()).expect("Can't parse the input constant_bytes into FieldElement");
-        Some(&constant_fe)
+        match FieldElement::deserialize(constant_bytes.as_slice()) {
+            Ok(constant_fe) => Option::Some(constant_fe),
+            Err(_) => Option::None
```

Comment:
This is not correct, this is a failure and we shouldn't let execution continue. Return false.

Reply:
Done